packages/eth-rpc-cache/error.ts

```diff
@@ -0,0 +1,22 @@
+export class JsonRpcError extends Error {
+  constructor(
+    message: string,
+    public code: number
+  ) {
+    super(message)
+    this.code = code
+
+    // Set the prototype explicitly.
+    // See https://github.com/microsoft/TypeScript-wiki/blob/81fe7b91664de43c02ea209492ec1cea7f3661d0/Breaking-Changes.md#extending-built-ins-like-error-array-and-map-may-no-longer-work
+    Object.setPrototypeOf(this, JsonRpcError.prototype)
+  }
+}
+
+export const errors = {
+  internalServerError: (err: Error) =>
+    new JsonRpcError(
+      err?.message && err.message !== '' ? err.message : 'Internal error',
```

Comment:
How is that different from this:

```suggestion
      err?.message || 'Internal error',
```

Reply:
![](https://64.media.tumblr.com/dc923c1b6f1f6e38fcea9beda25a9960/9b5e96271c48d355-13/s540x810/822704121ecdb4ee2a71511f899ea2942595576d.gifv)

fixed here [72a2ab2](https://github.com/bloq/eth-packages/pull/1/commits/72a2ab224eab60b727cac9dabc96dd48d23ad2c2)